### Assignment 4 (12th September 2022, Monday)


#### Initializing Connection


In [1]:
import mysql.connector as sqltor

connection = sqltor.connect(host="localhost", user="root", password="mysqlpasswd")
cursor = connection.cursor()
cursor.execute("CREATE DATABASE IF NOT EXISTS week4")
cursor.execute("USE week4")


#### Adding Helper functions


In [2]:
def disp():
    print(*[line for line in cursor], sep="\n")


def run(query: str):
    cursor.execute(query)


#### Testing Tables and Datasets


In [3]:
run("SHOW DATABASES")
disp()
run("SHOW TABLES")
disp()
# exit()


('information_schema',)
('mysql',)
('performance_schema',)
('sys',)
('week1',)
('week2',)
('week3',)
('week3i',)
('week4',)



| Database           |
| ------------------ |
| information_schema |
| mysql              |
| performance_schema |
| sys                |
| week1              |
| week2              |
| week3              |
| week3i             |
| week4              |


#### Queries


##### 1. Show the use of upper and lower function.


In [4]:
run("SELECT UPPER('hello world') AS UPPER")
disp()
run("SELECT LOWER('HELLO WORLD') AS LOWER")
disp()


('HELLO WORLD',)
('hello world',)


TABLE PREVIEW


| UPPER       |
| :---------- |
| HELLO WORLD |

| LOWER       |
| :---------- |
| hello world |


##### 2. Show the use of concat, instr and length function.


In [5]:
run("SELECT CONCAT('HELLO', 'WORLD') AS CONCATENATE")
disp()
run("SELECT INSTR('Pritam A40', 'A') AS 'FIND STRING'")
disp()


('HELLOWORLD',)
(5,)


TABLE PREVIEW


| CONCATENATE |
|:-|
| HELLOWORLD  |

| FIND STRING |
|:-|
|           5 |

##### 3. Show the use of the following functions on numeric values:

    i. Sqrt()
    ii. Power()
    iii. Ceil()
    iv. Substr()
    v. Max()
    vi. Min()
    vii. Round()
    viii. Avg()
    ix. Count()
    x. Exp()
    xi. Mod()


PRE REQUISITE


In [6]:
run("CREATE TABLE IF NOT EXISTS INTEGERS(NUMBER INT)")
run("TRUNCATE INTEGERS")
run("INSERT INTO INTEGERS VALUES (1), (2), (3), (4), (5)")
connection.commit()


QUERIES


In [ ]:
run("SELECT SQRT(1024) AS SQRT")
disp()
run("SELECT POWER(4, 2) AS POWER")
disp()
run("SELECT CEIL(4.9) AS CEIL")
disp()
run("SELECT SUBSTR('PRITAM KUNDU', 1, 6) AS SUBSTRING")
disp()
run("SELECT MAX(NUMBER) AS MAXIMUM FROM INTEGERS")
disp()
run("SELECT MIN(NUMBER) AS MINIMUM FROM INTEGERS")
disp()
run("SELECT ROUND(9.98) AS ROUND")
disp()
run("SELECT AVG(NUMBER) AS AVERAGE FROM INTEGERS")
disp()
run("SELECT COUNT(*) AS `NUMBER OF INTEGERS` FROM INTEGERS")
disp()
run("SELECT EXP(3) AS EXPONENT")
disp()
run("SELECT MOD(7, 3) AS MODULUS")
disp()


TABLE PREVIEW


| SQRT |
|:-|
|   32 |

| POWER |
|:-|
|    16 |

##### 4. Solve the following queries

    1. Find the ceiling and floor value of 14.887
    2. Find out the round off of 17.49989
    3. Calculate 8<sup>7</sup>


In [43]:
run("SELECT CEIL(14.887), FLOOR(14.887)"); disp()
run("SELECT ROUND(17.49989)")
disp()
run("SELECT POW(8, 7)")
disp()


(15, 14)
(Decimal('17'),)
(2097152.0,)


TABLE PREVIEW


| CEIL(14.887) | FLOOR(14.887) |
|-|-|
|           15 |            14 |


| ROUND(17.49989) |
|-|
|              17 |

| POW(8, 7) |
|-|
|   2097152 |

##### 5. Show the current date


In [45]:
run("SELECT CURDATE()")
disp()


(datetime.date(2022, 9, 13),)


TABLE PREVIEW


| CURDATE()  |
|-|
| 2022-10-16 |

##### 6. Find the total experience of the employees in weeks who works in Sales department


In [ ]:
run(
    """
    SELECT SUM(Experience) FROM Employees 
    WHERE Dept = 'Sales' GROUP BY WEEK
"""
)
disp()


(datetime.date(2022, 9, 13),)


TABLE PREVIEW


|SUM(Experience)|
|-:|
|100|
|20|
|40|

##### 7. Display the use of the following functions on date

    1. Months_between
    2. Add_months
    3. Next_day
    4. last_day
    5. Round
    6. Trunc
    7. To_char


In [66]:
run("SELECT TIMESTAMPDIFF(month, '2021-05-18', '2021-07-29') AS MONTH_DIFF")
disp()
run("SELECT DATE_ADD('2002-03-15', INTERVAL 1 MONTH) AS MONTH_ADD")
disp()
run("SELECT DATE_ADD(NOW(), INTERVAL 1 DAY)")
disp()
run("SELECT DATE_SUB(NOW(), INTERVAL 1 DAY)")
disp()
# 5
# 6
run("SELECT DATE_FORMAT(NOW(), '%D-%M-%Y')")
disp()


(2,)
('2002-04-15',)
(datetime.datetime(2022, 9, 14, 15, 19, 14),)
(datetime.datetime(2022, 9, 12, 15, 19, 14),)
('13th-September-2022',)


TABLE PREVIEW


| MONTH_DIFF |
|-|
|                                                2 |

| MONTH_ADD   |
|-|
| 2002-04-15 |

##### 8. Show the employee details with a revised salary. The salary is incremented in the following way:

    1. 10% for sales department
    2. 20% for marketing department
    3. No increment for others


In [ ]:
run("SELECT CURDATE()")
disp()


(datetime.date(2022, 9, 13),)


TABLE PREVIEW


In [62]:
disp()


('2002-04-15',)


##### 9. Determine the tax for each employee in production department based on the monthly salary. The tax rate are as per the following data:

| Monthly Salary Range | Rate |
| :------------------- | :--- |
| 0-19,999             | 0%   |
